flink-core/src/main/java/org/apache/flink/api/common/state/StateTtlConfig.java

```diff
@@ -71,32 +73,61 @@
 	}
 
 	/**
+	 * @deprecated use {@link TtlTimeCharacteristic} to avoid class name clash
+	 * with <code>org.apache.flink.streaming.api.TimeCharacteristic</code>.
+	 *
 	 * This option configures time scale to use for ttl.
 	 */
+	@Deprecated
 	public enum TimeCharacteristic {
 		/** Processing time, see also <code>TimeCharacteristic.ProcessingTime</code>. */
 		ProcessingTime
 	}
 
+	/**
+	 * This option configures time scale to use for ttl.
+	 */
+	public enum TtlTimeCharacteristic {
+		/** Processing time, see also <code>org.apache.flink.streaming.api.TimeCharacteristic.ProcessingTime</code>. */
+		ProcessingTime
+	}
+
 	private final UpdateType updateType;
 	private final StateVisibility stateVisibility;
-	private final TimeCharacteristic timeCharacteristic;
+	@Deprecated
+	private TimeCharacteristic timeCharacteristic;
```

Comment:
could we get rid of this field, also in Builder? `TimeCharacteristic` can be always converted into `TtlTimeCharacteristic` and back in relevant methods.